gubernator/templates/build.html

```diff
@@ -25,14 +25,20 @@
 <div id="failures">
 % if failures
 <h2>Test Failures</h2>
-% for name, time, text in failures
+<hr>
+% for name, time, text, filename in failures
 <a class="anchor" id="{{name|slugify}}" href="#{{name|slugify}}"><h3>{{name}}<span class="time"> {{time|duration}}</span></h3></a>
 <pre class="cmd" onclick="select(this)">{{name | testcmd}}</pre>
 % if text
 <pre class="error">
 {{text|linkify_stacktrace(commit)}}
 </pre>
 % endif
+% if pods[name, time, text, filename]
```

Comment:
You could do the pod name extraction here, like linkify_stacktrace does for commits. This might look like:

    % set pod_name = parse_pod_name(text)
    % if pod_name
    ... <a href="/build{{build_dir}/nodelog?pod={{pod_name}}&junit=filename"> ...
    %endif